frontends/common/options.cpp

```diff
@@ -130,6 +133,38 @@ void CompilerOptions::setInputFile() {
     }
 }
 
+std::vector<const char*>* CompilerOptions::process(int argc, char* const argv[]) {
+    char buffer[PATH_MAX];
+    int len;
+    struct stat st;
+    if ((len = readlink("/proc/self/exe", buffer, sizeof(buffer))) > 0 ||
```

Comment:
If I understand correctly this code is finding the directory from where the program was invoked. Why not use dirname(argv[0])? It is much more portable than using /proc. This implementation will not work on MacOS.